pkg/operator/injection/injection.go

```diff
@@ -57,3 +63,29 @@ func WithOptionsOrDie(ctx context.Context, opts ...options.Injectable) context.C
 	}
 	return ctx
 }
+
+func WithClient(ctx context.Context, client client.Client) context.Context {
```

Comment:
This feels tenuous to me since it enforces an anti-pattern IMO of injecting the client into the context. Does knative already directly support this? Are there no other ways around doing something like this?

Reply:
I personally don't love this solution my self. Knative does not provide support for accessing the client or accessing other resource inside of the cluster. To minimize the context injection, we are only going to be containing the client on the webhooks side.

Reply:
Do we need this injection anymore now that we don't have cross-resource access in our conversion webhooks?

Reply:
I was initially thinking we might have drift issue and needed to pull v1 annotations, however after I tested the hash controller can handle the changes that users make to cause drift

Reply:
+1